src/NUnitFramework/framework/Constraints/UniqueItemsConstraint.cs

```diff
@@ -60,5 +69,123 @@ protected override bool Matches(IEnumerable actual)
 
             return true;
         }
+
+        private bool? TryFastAlgorithm(IEnumerable actual)
+        {
+            // If the user specified any external comparer with Using, exit
+            if (UsingExternalComparer)
+                return null;
+
+            // If IEnumerable<T> is not implemented exit,
+            // Otherwise return value is the Type of T
+            Type memberType = GetGenericTypeArgument(actual);
+            if (memberType == null || IsHandledSpeciallyByNUnit(memberType))
+                return null;
+
+            string methodName = nameof(ItemsUnique);
+            bool makeGeneric = true;
+
+            // Special handling for ignore case with strings and chars
+            if (IgnoringCase)
+            {
+                if (memberType == typeof(string))
+                {
+                    methodName = nameof(StringsUniqueIgnoringCase);
+                    makeGeneric = false;
+                }
+                else
+                if (memberType == typeof(char))
+                {
+                    methodName = nameof(CharsUniqueIgnoringCase);
+                    makeGeneric = false;
+                }
```

Comment:
I'd be much happier (and the code is faster) not using reflection for these two cases. Also there's a newline between `else` and `if`.

Reply:
Let me take a look again. An earlier version required it but I guess I got rid of that need and didn't notice. That said, there's a limit on how much time I want to spend on this one and  the slower code is about 100 times faster than what we had before.